packages/react-storage/src/components/StorageBrowser/Views/Controls/Download.tsx

```diff
@@ -42,18 +43,39 @@ const DownloadButton: typeof ButtonElement = React.forwardRef(
   }
 );
 
-export const DownloadControl: DownloadControl = ({ key }) => {
-  const [, handleDownload] = useAction({
-    type: 'DOWNLOAD',
+function download(fileName: string, url: string) {
+  const a = document.createElement('a');
+
+  a.href = url;
+  a.download = fileName;
+
+  document.body.appendChild(a);
+
+  a.click();
+
+  document.body.removeChild(a);
+}
+
+export const DownloadControl: DownloadControl = ({ fileKey }) => {
+  const [{ data }, handleDownload] = useDataState(downloadAction, {
+    signedUrl: '',
   });
```

Comment:
Using `useDataState` here instead because using context state was causing all the buttons in the table to re-render when I clicked a specific row to download. I think this encapsulates the state into each component individually